core/src/components/input/input.tsx

```diff
@@ -471,6 +472,21 @@ export class Input implements ComponentInterface {
     return typeof this.value === 'number' ? this.value.toString() : (this.value || '').toString();
   }
 
+  private getLabelPlacement() {
+    const theme = getIonTheme(this);
+    const { el, labelPlacement } = this;
+
+    if (theme === 'ionic' && labelPlacement !== 'stacked' && labelPlacement !== 'floating') {
```

Comment:
~ I don't know if we need the overhead for runtime checks for every unsupported prop/prop option for a given theme. 

Open to team feedback on this, but this feels like a problem to solve closer to beta/rc/ga. We may fill these feature gaps in that time and the documentation may be clear enough for developers to navigate.

Reply:
I'm open to removing the check and/or warning, but it was included in the design doc and is a pattern we're already using for other features. The `size` prop on `ion-input` is already implemented this way, for example.